### PR TITLE
implement format/expression in input model band definition

### PR DESCRIPTION
**Pull request by fmigneault**
## Description

Allow `mlm:input` to define `bands` with object representation:
```json
{
  "name": "band-name",
  "format": "expr-fmt",
  "expression": "{}"
}
```

Using this, the model can define per-band pre-processing steps, such as, computing a virtual NDVI band consisting of an arithmetic combination of other band sources.

## Related Issue

- Fixes https://github.com/crim-ca/mlm-extension/discussions/7

## Type of Change

- [x] :books: Examples, docs, tutorials or dependencies update;
- [ ] :wrench: Bug fix (non-breaking change which fixes an issue);
- [x] :clinking_glasses: Improvement (non-breaking change which improves an existing feature);
- [x] :rocket: New feature (non-breaking change which adds functionality);
- [ ] :boom: Breaking change (fix or feature that would cause existing functionality to change);
- [ ] :closed_lock_with_key: Security fix.

## Checklist

- [x] I've read the [`CONTRIBUTING.md`](../CONTRIBUTING.md) guide;
- [x] I've updated the code style using `make codestyle`;
- [x] I've written tests for all new methods and classes that I created;
- [x] I've written the docstring in `Google` format for all the methods and classes that I used.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -269,6 +272,29 @@ properties of the model.
...
+#### Model Band Object
+
+| Field Name | Type   | Description                                                                                                                 |
+|------------|--------|-----------------------------------------------------------------------------------------------------------------------------|
+| name       | string | **REQUIRED** Name of the band referring to an extended band definition (see [Bands](#bands-and-statistics).                 |
+| format     | string | The type of expression that is specified in the `expression` property.                                                      |
+| expression | \*     | An expression compliant with the `format` specified. The expression can be any data type and depends on the `format` given. |
+
+> :information_source: <br>
+> Although `format` and `expression` are not required in this context, they are mutually dependent of each other. <br>
```
Comment (rbavery) [inline README.md:284]:
```suggestion
> Although `format` and `expression` are not required in this context, they are mutually dependent on each other. <br>
```

File: README.md [hunk 2]
```diff
@@ -269,6 +272,29 @@ properties of the model.
 [stac-raster-stats]: https://github.com/stac-extensions/raster?tab=readme-ov-file#statistics-object
 [stac-band-names]: https://github.com/stac-extensions/eo?tab=readme-ov-file#common-band-names
 
+#### Model Band Object
+
+| Field Name | Type   | Description                                                                                                                 |
+|------------|--------|-----------------------------------------------------------------------------------------------------------------------------|
+| name       | string | **REQUIRED** Name of the band referring to an extended band definition (see [Bands](#bands-and-statistics).                 |
+| format     | string | The type of expression that is specified in the `expression` property.                                                      |
+| expression | \*     | An expression compliant with the `format` specified. The expression can be any data type and depends on the `format` given. |
```
Comment (rbavery) [inline README.md:281]: Was initially a bit confused reading this. Should it be rephrased to 

```suggestion
| expression | \*     | An expression compliant with the `format` specified. The expression can be applied to any numerical data type and depends on the `format` given. |
```

**Review by rbavery — CHANGES_REQUESTED** [review]
Looks great! Just some minor language edits, tests and descriptions are thorough.

File: README.md (hunk 2 above)
Comment (fmigneault) [reply README.md:281]: Yes for the "applied to". 
I think we should avoid the "numerical" restriction. Not sure if there would realistically be any non-numeric cases, but I prefer to leave the flexibility.

Comment (rbavery) [reply README.md]: ok sounds good, agreed

**Review by rbavery — APPROVED** [review]

**Pull request merged**